code/src/test/kotlin/lexicalTests/LexerTest.kt

```diff
@@ -76,6 +74,176 @@ class LexerTest {
         assertThat(positionalToken.lineNumber, equalTo(ExpectedLineNumber))
     }
 
+    @org.junit.jupiter.api.Test
+    fun lexerTestTypeDeclarationTokens() {
+        val lex = Lexer("var none num func tuple list bool txt")
+
+        val tokens = lex.getTokenSequence().toList()
+
+        assertTrue(tokens[0].token is lexer.Token.Type.Var)
+        assertTrue(tokens[1].token is lexer.Token.Type.None)
+        assertTrue(tokens[2].token is lexer.Token.Type.Number)
+        assertTrue(tokens[3].token is lexer.Token.Type.Func)
+        assertTrue(tokens[4].token is lexer.Token.Type.Tuple)
+        assertTrue(tokens[5].token is lexer.Token.Type.List)
+        assertTrue(tokens[6].token is lexer.Token.Type.Bool)
+        assertTrue(tokens[7].token is lexer.Token.Type.Text)
+    }
+
+    @org.junit.jupiter.api.Disabled
+    @org.junit.jupiter.api.Test
+    fun lexerTestLegalIdentifiers() {
+        val lex = Lexer("hej var1 +- æøå ↑ ☺☻ ??? _this £50 *bold* 6lol")
+
+        val tokens = lex.getTokenSequence().toList()
+
+        tokens.dropLast(1).forEach{ assertTrue(it.token is lexer.Token.Identifier) }
+        assertTrue(tokens.last().token is lexer.Token.SpecialChar.EndOfLine)
+    }
+
+    @org.junit.jupiter.api.Disabled
+    @org.junit.jupiter.api.Test
+    fun lexerTestIllegalIdentifiers() {
+        val lex = Lexer("var = \"id \"id\" 'di 'di' [] { () true \\ , }")
+        //Tokens: var, =, "id, "id", 'di, 'di', [, ], {, (, ), true, \, ",", ->, }
+        val tokens = lex.getTokenSequence().toList()
+
+        tokens.dropLast(1).forEach{ assertTrue(it.token !is lexer.Token.Identifier) }
+        assertTrue(tokens.last().token is lexer.Token.SpecialChar.EndOfLine)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestBooleanLiteralTokens() {
+        val lex = Lexer("true false")
+
+        val tokens = lex.getTokenSequence().toList()
+
+        assertTrue(tokens[0].token is lexer.Token.Literal.Bool)
+        assertTrue(tokens[1].token is lexer.Token.Literal.Bool)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestFunctionDeclaration() {
+        val lex = Lexer("func print = (txt message) : none\n{ }")
```

Comment:
we need to test that =( is two symbols as well. maybe in general that no-white-space works with special chars in combo with both identifiers and types - there is a method that does this for identifiers already. that robably just needs to be extended